### PR TITLE
New material on formation sequences for first-order logic and propositional logic

### PR DESCRIPTION
**Pull request by beastaugh**
New material on formation sequences for first-order logic and propositional logic, based on what can be found in Smullyan and more recent textbooks like van Dalen.

### CODE REVIEW

**Comment by beastaugh** [comment]
Had to fix some compile errors, hence the force push (it was working for me locally but the CI must be loading some package that does something with underscores that my local setup doesn't). Should be working ok now.

**Comment by rzach** [comment]
Will have look soon. (Underscores are active (subscripts!) so they don't work outside math.)

File: content/first-order-logic/syntax-and-semantics/subformulas.tex [hunk 1]
```diff
@@ -87,4 +87,25 @@
...
 
+\begin{prop}
+\ollabel{prop:subfrm-trans}
+Suppose $!B$ is a subformula of $!A$ and $!C$ is a subformula of $!B$.
+Then $!C$ is a subformula of $!A$. In other words, the subformula
+relation is transitive.
+\end{prop}
+
+\begin{prob}
+Prove \olref{prop:subfrm-trans}.
```
Comment (rzach) [inline content/first-order-logic/syntax-and-semantics/subformulas.tex:98]: Can you put full references here (\olref[..][..][..]{..})?

**Review by rzach — CHANGES_REQUESTED** [review]

File: content/first-order-logic/syntax-and-semantics/subformulas.tex (hunk 1 above)
Comment (rzach) [reply content/first-order-logic/syntax-and-semantics/subformulas.tex:98]: (and below)

Comment (beastaugh) [reply content/first-order-logic/syntax-and-semantics/subformulas.tex]: Done in [f0e48d7](https://github.com/OpenLogicProject/OpenLogic/pull/317/commits/f0e48d7b69fd4f138fac1077ae74a5a5d3590fca).

**Review by rzach — APPROVED** [review]

**Comment by rzach** [comment]
Thanks!

**Comment by rzach** [comment]
(I made a few changes fyi)

**Comment by beastaugh** [comment]
Looks good, thanks!

**Pull request merged**